tests/testDoubleShapeletPsfApprox.py

```diff
@@ -0,0 +1,398 @@
+        refRecord = refCat.addNew()
+        refRecord.set(refCentroidKey, lsst.afw.geom.Point2D(0.0, 0.0))
+        refWcs = self.exposure.getWcs() # same as measurement Wcs
+        task = lsst.meas.base.ForcedMeasurementTask(config=config, refSchema=refSchema)
+        measCat = task.generateMeasCat(self.exposure, refCat, refWcs)
+        task.run(measCat, self.exposure, refCat, refWcs)
+        measRecord = measCat[0]
+        self.assertFalse(measRecord.get("modelfit_DoubleShapeletPsfApprox_flag"))
+        measSchema = measCat.schema
+        key = lsst.shapelet.MultiShapeletFunctionKey(measSchema["modelfit"]["DoubleShapeletPsfApprox"])
+        msf = measRecord.get(key)
+        self.checkBounds(msf)
+        self.checkFitQuality(msf)
+
+    def testInitializeResult(self):
+        """Test that initializeResult() returns a unit-flux, unit-circle MultiShapeletFunction
+        with the right peakRatio and radiusRatio.
+        """
+        msf = self.Algorithm.initializeResult(self.ctrl)
+        self.assertClose(msf.evaluate().integrate(), 1.0)
+        moments = msf.evaluate().computeMoments()
+        axes = lsst.afw.geom.ellipses.Axes(moments.getCore())
+        self.assertClose(moments.getCenter().getX(), 0.0)
+        self.assertClose(moments.getCenter().getY(), 0.0)
+        self.assertClose(axes.getA(), 1.0)
+        self.assertClose(axes.getB(), 1.0)
+        self.assertEqual(len(msf.getComponents()), 2)
+        self.checkRatios(msf)
+
+    def testFitMoments(self):
+        """Test that fitMoments() preserves peakRatio and radiusRatio while setting moments
+        correctly.
+        """
+        image = self.psf.computeKernelImage()
+        array = image.getArray()
+        bbox = image.getBBox()
+        x, y = numpy.meshgrid(
+            numpy.arange(bbox.getBeginX(), bbox.getEndX()),
+            numpy.arange(bbox.getBeginY(), bbox.getEndY())
+        )
+        msf = self.Algorithm.initializeResult(self.ctrl)
+        self.Algorithm.fitMoments(msf, self.ctrl, image)
+        self.assertClose(msf.evaluate().integrate(), array.sum())
+        moments = msf.evaluate().computeMoments()
+        q = lsst.afw.geom.ellipses.Quadrupole(moments.getCore())
+        cx = (x*array).sum()/array.sum()
+        cy = (y*array).sum()/array.sum()
+        self.assertClose(moments.getCenter().getX(), cx)
+        self.assertClose(moments.getCenter().getY(), cy)
+        self.assertClose(q.getIxx(), ((x - cx)**2 * array).sum()/array.sum(), rtol=1E-14)
+        self.assertClose(q.getIyy(), ((y - cy)**2 * array).sum()/array.sum(), rtol=1E-14)
+        self.assertClose(q.getIxy(), ((x - cx)*(y - cy)*array).sum()/array.sum(), rtol=1E-14)
+        self.assertEqual(len(msf.getComponents()), 2)
+        self.checkRatios(msf)
+        self.checkBounds(msf)
+
+    def testObjective(self):
+        """Test that model and derivative evaluation in the objective object provided
+        to the optimizer when fitting the profile.
```

Comment:
Fixed; should have been "Test that model evaluation agrees with derivative evaluation".